server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -243,7 +245,12 @@ PublishWithJoinResponse handleIncomingRemotePublishRequest(RemotePublishRequest
         }
         boolean applyFullState = false;
         final ClusterState lastSeen = lastSeenClusterState.get();
-        if (lastSeen == null) {
+        if (remoteClusterStateService.getRemotePublicationApplyFullState()) {
```

Comment:
if a node has stale cluster state and this setting was changed from true to false, it will try to find diff and it would fail, would there be exception and crash OpenSearch process?

Reply:
We have a check to ensure that we only try to apply if it has been published following this piece of code, if we haven't published the diff we will download and apply the full state only.